### ISSUE TITLE
v2022.08.02 - panic at startup on fresh sync

### ISSUE THREAD

**Issue by MysticRyuujin** [ledgerwatch/erigon#5096]
#### System information

Erigon version: `erigon version 2022.08.2-alpha-35c4faa1`

```
[INFO] [08-17|16:44:46.803] Build info                               git_branch=HEAD git_tag=v2022.08.02 git_commit=35c4faa1b41e8379a74d0385505add0dd450c2ed
[INFO] [08-17|16:44:46.804] Starting Erigon on Ethereum mainnet... 
[INFO] [08-17|16:44:46.804] Maximum peer count                       ETH=100 total=100
[INFO] [08-17|16:44:46.804] starting HTTP APIs                       APIs=eth,admin,debug,net,trace,web3,erigon,engine
[INFO] [08-17|16:44:46.804] torrent verbosity                        level=WRN
[INFO] [08-17|16:44:48.905] Set global gas cap                       cap=300000000
[INFO] [08-17|16:44:48.939] Opening Database                         label=chaindata path=/data/erigon/datadir/chaindata
[INFO] [08-17|16:44:48.940] Initialised chain configuration          config="{ChainID: 1, Homestead: 1150000, DAO: 1920000, DAO Support: true, Tangerine Whistle: 2463000, Spurious Dragon: 2675000, Byzantium: 4370000, Constantinople: 7280000, Petersburg: 7280000, Istanbul: 9069000, Muir Glacier: 9200000, Berlin: 12244000, London: 12965000, Arrow Glacier: 13773000, Gray Glacier: 15050000, Terminal Total Difficulty: 58750000000000000000000, Merge Netsplit: <nil>, Engine: ethash}" genesis=0xd4e56740f876aef8c010b86a40d5f56745a118d0906a34e69aec8c0db1cb8fa3
[INFO] [08-17|16:44:48.940] Effective                                prune_flags= snapshot_flags="--snapshots=true" history.v2=false
[INFO] [08-17|16:44:51.056] Initialising Ethereum protocol           network=1
[INFO] [08-17|16:44:51.056] Disk storage enabled for ethash DAGs     dir=/data/erigon/ethash count=2
[INFO] [08-17|16:44:51.058] Starting private RPC server              on=127.0.0.1:9090
[INFO] [08-17|16:44:51.058] new subscription to logs established 
[INFO] [08-17|16:44:51.058] rpc filters: subscribing to Erigon events 
[INFO] [08-17|16:44:51.058] New txs subscriber joined 
[EROR] [08-17|16:44:51.058] catch panic                              err="runtime error: invalid memory address or nil pointer dereference" stack="[main.go:29 panic.go:838 panic.go:220 signal_unix.go:818 index.go:149 block_snapshots.go:179 config.go:245 block_snapshots.go:286 config.go:243 backend.go:572 node.go:115 main.go:58 app.go:526 app.go:286 main.go:34 proc.go:250 asm_amd64.s:1571]"
```

**Comment by MysticRyuujin** [comment]
Hmmm this could potentially be because I kept the snapshots from a previous sync, and did not delete/remove the `db` folder within the snapshots folder...?

**Comment by 3nprob** [comment]
Looks like duplicate of #5068

**Comment by 3nprob** [comment]
Seems like erigon is saving corrupted files.

Did not dig deeper but perhaps this helps in debugging and unblocking: Running the process with `strace erigon ... | grep -E 'snapshots/'` will output which files are being read by the process. Then, if the last line before the process dies is:

```
openat(AT_FDCWD, "/mydir/snapshots/v1-007000-007500-transactions.seg.torrent")
```

, I would delete `/mydir/snapshots/v1-007000-007500-transactions.*`and try again.

After a couple of times (I ended up moving ~10% of the 200GB synced so far), the process continues startup and snapshot sync starts up again, reporting 95% progress (which seems sus considering that more than that was removed, will see what happens when it completes).
